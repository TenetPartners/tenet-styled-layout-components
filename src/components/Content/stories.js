/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf, addDecorator } from '@storybook/react';
import { text, withKnobs } from '@storybook/addon-knobs';
import Content, { HTMLContent } from '.';

addDecorator(withKnobs);

storiesOf('Content', module)
  .add('with regular content as a label', () => {
    const content = text('Content', 'hello world');
    return (
      <Content content={content} />
    )
  })
  .add('with css class', () => {
    const content = text('Content', 'hello world');
    const cssclass = text('CSS class', 'test');
    return (
      <Content content={content} className={cssclass} />
    )
  });

storiesOf('HTML Content', module)
  .add('with HTML content', () => (
    <HTMLContent content="<p>hello</p><p>new paragraph</p>" />
  ))
  .add('with css class', () => {
    const content = text('Content', 'hello world');
    const cssclass = text('CSS class', 'test');
    return (
      <HTMLContent content={content} className={cssclass} />
    )
  })
  .add('with different tag', () => {
    const content = text('Content', 'hello world');
    const tag = text('Tag', 'h2');
    return (
      <HTMLContent content={content} tag={tag}/>
    );
  });
