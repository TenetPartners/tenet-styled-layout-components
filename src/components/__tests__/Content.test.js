/* eslint-disable function-paren-newline */
import React from 'react';
import renderer from 'react-test-renderer';
import Content, { HTMLContent } from '../Content';

describe('Content', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Content content="<p>hello</p>" />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    const div = tree.root.findByType('div');
    expect(div.children).toEqual(['<p>hello</p>']);
  });

  it('renders correctly given a class', () => {
    const tree = renderer.create(
      <Content content="<p>hello</p>" className="asdf" />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    const div = tree.root.findByType('div');
    expect(div.props.className).toBe('asdf');
  });
});

describe('HTMLContent', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <HTMLContent content="<p>hello</p>" />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    const div = tree.root.findByType('div');
    expect(div.props.dangerouslySetInnerHTML).toEqual({ __html: '<p>hello</p>' });
    expect(div.children).toEqual([]);
  });

  it('renders correctly given a class', () => {
    const tree = renderer.create(
      <HTMLContent content="<p>hello</p>" className="asdf" />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    const div = tree.root.findByType('div');
    expect(div.props.className).toBe('asdf');
  });

  it('renders the tag provided', () => {
    const tree = renderer.create(
      <HTMLContent content="<p>hello</p>" tag="h3" />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.root.findByType('h3')).toBeDefined();
  });
});
