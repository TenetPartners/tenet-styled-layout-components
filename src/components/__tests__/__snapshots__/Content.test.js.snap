// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div>
  &lt;p&gt;hello&lt;/p&gt;
</div>
`;

exports[`Content renders correctly given a class 1`] = `
<div
  className="asdf"
>
  &lt;p&gt;hello&lt;/p&gt;
</div>
`;

exports[`HTMLContent renders correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>hello</p>",
    }
  }
/>
`;

exports[`HTMLContent renders correctly given a class 1`] = `
<div
  className="asdf"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>hello</p>",
    }
  }
/>
`;

exports[`HTMLContent renders the tag provided 1`] = `
<h3
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>hello</p>",
    }
  }
/>
`;
